src/Microsoft.AspNetCore.Sockets.Abstractions/ConnectionContext.cs

```diff
@@ -0,0 +1,30 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Security.Claims;
+using System.Text;
+using System.Threading;
+using Microsoft.AspNetCore.Http.Features;
+
+namespace Microsoft.AspNetCore.Sockets
+{
+    public abstract class ConnectionContext : IDisposable
+    {
+        public abstract string ConnectionId { get; }
+
+        public abstract IFeatureCollection Features { get; }
+
+        public abstract ClaimsPrincipal User { get; set; }
+
+        // REVIEW: Should this be changed to items
```

Comment:
I don't want to do that in this change. The main thing here is the assembly changes. I can file an issue to follow up (like getting rid of Metatdata altogether).